desc/magnetic_fields.py

```diff
@@ -1094,3 +1130,644 @@ def odefun(rpz, s):
+        return self._I
+
+    @I.setter
+    def I(self, new):  # noqa: E743
+        assert np.isscalar(new), "I must be a scalar"
+        self._I = new
+
+    @property
+    def G(self):
+        """Net current linking the plasma and the coils poloidally."""
+        return self._G
+
+    @G.setter
+    def G(self, new):
+        assert np.isscalar(new), "G must be a scalar"
+        self._G = new
+
+    @property
+    def Phi_mn(self):
+        """Fourier coefficients describing single-valued part of potential."""
+        return self._Phi_mn
+
+    @Phi_mn.setter
+    def Phi_mn(self, new):
+        if len(new) == self.Phi_basis.num_modes:
+            self._Phi_mn = jnp.asarray(new)
+        else:
+            raise ValueError(
+                f"Phi_mn should have the same size as the basis, got {len(new)} for "
+                + f"basis with {self.Phi_basis.num_modes} modes."
+            )
+
+    @property
+    def Phi_basis(self):
+        """DoubleFourierSeries: Spectral basis for Phi."""
+        return self._Phi_basis
+
+    @property
+    def sym_Phi(self):
+        """str: Type of symmetry of periodic part of Phi (no symmetry if False)."""
+        return self._sym_Phi
+
+    def change_Phi_resolution(self, *args, **kwargs):
+        """Change the maximum poloidal and toroidal resolution for Phi."""
+        assert (
+            ((len(args) in [2, 3]) and len(kwargs) == 0)
+            or ((len(args) in [2, 3]) and len(kwargs) in [1, 2])
+            or (len(args) == 0)
+        ), (
+            "change_Phi_resolution should be called with 2 (M,N) or 3 (L,M,N) "
+            + "positional arguments or only keyword arguments."
+        )
+        L = kwargs.pop("L", None)
+        M = kwargs.pop("M", None)
+        N = kwargs.pop("N", None)
+        NFP = kwargs.pop("NFP", None)
+        sym_Phi = kwargs.pop("sym_Phi", None)
+        assert len(kwargs) == 0, "change_resolution got unexpected kwarg: {kwargs}"
+        self._NFP = NFP if NFP is not None else self.NFP
+        self._sym_Phi = sym_Phi if sym_Phi is not None else self.sym_Phi
+        if L is not None:
+            warnings.warn("Phi basis does not have radial resolution, ignoring L")
+        if len(args) == 2:
+            M, N = args
+        elif len(args) == 3:
+            L, M, N = args
+
+        if (
+            ((N is not None) and (N != self.N))
+            or ((M is not None) and (M != self.M))
+            or (NFP is not None)
+        ):
+            M = M if M is not None else self.M
+            N = N if N is not None else self.N
+            Phi_modes_old = self.Phi_basis.modes
+            self.Phi_basis.change_resolution(M=M, N=N, NFP=self.NFP, sym=self.sym_Phi)
+
+            self._Phi_mn = copy_coeffs(self.Phi_mn, Phi_modes_old, self.Phi_basis.modes)
+            self._M_Phi = M
+            self._N_Phi = N
+
+    def compute_magnetic_field(self, coords, params=None, basis="rpz", grid=None):
```

Comment:
if we want to reduce duplicated code we could make this a private standalone function and then have both methods call that, similar to some of the `Equilibrium` methods